js/build.js

```diff
@@ -2636,6 +2635,22 @@ Fliplet.Widget.instance('chat', function (data) {
       });
   }
 
+  function getUserEmail() {
```

Comment:
this should first use `fl-chat-auth-email` so existing apps work as expected, then if `undefind` fallback to reading from the cached session

Reply:
Is this because you expect existing apps will have users who are logged in, but won't have a session? I assume the session data will always be available if a users is logged in therefore there's no need for backward compatibility.

Reply:
@tonytlwu are you managing the risk of potentially all apps using chat breaking because you've changed how this works?

I don't want to and I don't see the point of changing how this works unless there's a good reason. I'd simply add reading from the session as a fallback so that it works on portals.

Reply:
Yes there is a good reason. https://github.com/Fliplet/fliplet-studio/issues/5169 suggests that the authentication email isn't available across apps.

This is indeed changing how all chat apps get the user email, but neither of us seems to be able to find any reason why a user would be logged in and not have a session, i.e. has `fl-chat-auth-email` stored but no session data.

Reply:
@tonytlwu I still would keep the logic as is because it was built as such for the reason of the data source potentially being different from the one used to log in, or even use a different email for chat than the one used on the connected data source entry from the session (see my comment https://github.com/Fliplet/fliplet-studio/issues/5169#issuecomment-545454188).

There might be apps out there manually setting `fl-chat-auth-email` on certain occasions, why do we want to take such risk?

A better approach in my opinion is:
1. read from profile (or cached session) as fallback
2. update login component not to set `fl-chat-auth-email` so that we technically start deprecating it in favour or the user's profile
3. in the future, bump up the chat version to remove any read from `fl-chat-auth-email`; moreover, add a hook so that such value can manually be changed if required when the chat initialise.